src/java/org/apache/cassandra/db/marshal/CompositeType.java

```diff
@@ -222,8 +222,15 @@ protected <V> AbstractType<?> getAndAppendComparator(int i, V value, ValueAccess
     @Override
     public <V> ByteSource asComparableBytes(ValueAccessor<V> accessor, V data, Version version)
     {
-        if (data == null || accessor.isEmpty(data))
-            return null;
```

Comment:
This method is public. Why remove this check? Is it going to work with null data or empty accessor? 
If we're sure it should never accept null data / empty accessor, then maybe add a precondition instead?